CRM/Contact/Form/Contact.php

```diff
@@ -1128,13 +1135,16 @@ public static function blockDataExists(&$fields) {
    *
    * @param array $fields
    *   Fields array which are submitted.
-   * @param array $errors
    * @param int $contactID
    *   Contact id.
    * @param string $contactType
    *   Contact type.
+   *
+   * @return string|null
+   *
+   * @throws \CRM_Core_Exception
    */
-  public static function checkDuplicateContacts(&$fields, &$errors, $contactID, $contactType) {
+  private static function checkDuplicateContacts(array $fields, int $contactID, string $contactType): ?string {
```

Comment:
I thought this would be an easy PR but I should know by now there is no such thing.
1. This gives a type error when saving a new contact since contactID is null.
2. It only hits any of this code when you have an obscure setting changed from the default, which wasn't obvious at first. (Display Prefs - Check for Similar Contacts - On save).
3. There's still a check for empty() in the template, which I had understood was problematic for this smarty mode. Unless it was just that empty() kind of gets ignored in smarty mode since it evaluates the var anyway, so not so much problematic but doesn't add anything. If so, then the empty() could be removed.

Otherwise though, seems good.